packages/react-native-codegen/src/parsers/typescript/parseTopLevelType.js

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow strict-local
+ * @format
+ */
+
+'use strict';
+
+import type {TypeDeclarationMap} from './utils.js';
+
+export type LegalDefaultValues = string | number | boolean | null;
+
+type TopLevelTypeInternal = {
+  unions: Array<$FlowFixMe>,
+  optional: boolean,
+  defaultValue?: LegalDefaultValues,
+};
+
+export type TopLevelType = {
+  type: $FlowFixMe,
+  optional: boolean,
+  defaultValue?: LegalDefaultValues,
+};
+
+function getValueFromTypes(
+  value: $FlowFixMe,
+  types: TypeDeclarationMap,
+): $FlowFixMe {
+  switch (value.type) {
+    case 'TSTypeReference':
+      if (types[value.typeName.name]) {
+        return getValueFromTypes(types[value.typeName.name], types);
+      } else {
+        return value;
+      }
+    case 'TSTypeAliasDeclaration':
+      return getValueFromTypes(value.typeAnnotation, types);
+    default:
+      return value;
+  }
+}
+
+function isNull(t: $FlowFixMe) {
+  return (
+    t.type === 'TSNullKeyword' ||
+    t.type === 'TSUndefinedKeyword' ||
+    t.type === 'TSVoidKeyword'
+  );
+}
+
+function evaluateLiteral(
+  literalNode: $FlowFixMe,
+): string | number | boolean | null {
+  const valueType = literalNode.type;
+  if (valueType === 'TSLiteralType') {
+    const literal = literalNode.literal;
+    if (
+      literal.type === 'Literal' ||
+      literal.type === 'StringLiteral' ||
+      literal.type === 'NumericLiteral' ||
+      literal.type === 'BooleanLiteral'
+    ) {
+      if (
+        typeof literal.value === 'string' ||
+        typeof literal.value === 'number' ||
+        typeof literal.value === 'boolean'
+      ) {
+        return literal.value;
+      }
+    } else if (
+      literal.type === 'UnaryExpression' &&
+      literal.operator === '-' &&
+      (literal.argument.type === 'Literal' ||
+        literal.argument.type === 'NumericLiteral') &&
+      typeof literal.argument.value === 'number'
+    ) {
+      return -literal.argument.value;
+    }
```

Comment:
It could be `BigIntLiteral` according to [the document](https://babeljs.io/docs/en/babel-types). In this case, `literal.value` is a string. So "2." doesn't work. But I've extracted the type list to functions.

Reply:
Here I test `Literal` because, I don't know why but in `react-native-windows`, when I run babel, the result will be sometimes different from what the document says, by giving me an AST that you could get from `AST explorer` with `@typescript-eslint/parser`.